pyproject.toml

```diff
@@ -1,7 +1,76 @@
 [build-system]
-requires = ["setuptools>=42"]
+requires = ["setuptools>=61.2.0", "wheel"]
 build-backend = "setuptools.build_meta"
 
+[project]
+name = "lark"
+authors = [{name = "Erez Shinan", email = "erezshin@gmail.com"}]
+license = {text = "MIT"}
+description = "a modern parsing library"
+keywords = ["Earley", "LALR", "parser", "parsing", "ast"]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "Programming Language :: Python :: 3",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+    "Topic :: Text Processing :: General",
+    "Topic :: Text Processing :: Linguistic",
+    "License :: OSI Approved :: MIT License",
+]
+requires-python = ">=3.6"
+dependencies = []
+dynamic = ["version"]
+
+[project.readme]
+text = """
+Lark is a modern general-purpose parsing library for Python.
+With Lark, you can parse any context-free grammar, efficiently, with very little code.
+Main Features:
+- Builds a parse-tree (AST) automagically, based on the structure of the grammar
+- Earley parser
+- Can parse all context-free grammars
+- Full support for ambiguous grammars
+- LALR(1) parser
+- Fast and light, competitive with PLY
+- Can generate a stand-alone parser
+- CYK parser, for highly ambiguous grammars
+- EBNF grammar
+- Unicode fully supported
+- Automatic line & column tracking
+- Standard library of terminals (strings, numbers, names, etc.)
+- Import grammars from Nearley.js
+- Extensive test suite
+- And much more!
+Since version 1.0, only Python versions 3.6 and up are supported."""
+content-type = "text/markdown"
+
+[project.urls]
+Homepage = "https://github.com/lark-parser/lark"
+Download = "https://github.com/lark-parser/lark/tarball/master"
+
+[project.entry-points.pyinstaller40]
+hook-dirs = "lark.__pyinstaller:get_hook_dirs"
+
+[project.optional-dependencies]
+regex = ["regex"]
+nearley = ["js2py"]
+atomic_cache = ["atomicwrites"]
+
+[tool.setuptools]
```

Comment:
Hatching handles this better, IMO, and works even if some limits setuptools<60 in a non isolated environment (cough, NumPy, cough)

Reply:
FYI, from what I gather, the next version of setuptools is making `tool.setuptools` non-provisional.